packages/lending/src/adapter.ts

```diff
@@ -185,6 +185,299 @@ export class Adapter extends common.Web3Toolkit {
+        ).amount;
+        leverageOutput.logics[1].fields.balanceBps = common.BPS_BASE;
+      }
+      output.logics.push(...leverageOutput.logics);
+      output.afterPortfolio = leverageOutput.afterPortfolio;
+      output.destAmount = output.afterPortfolio.findBorrow(debtToken)!.balance;
+    } catch (err) {
+      output.error = err instanceof OperationError ? err : new OperationError('openByCollateral', 'UNEXPECTED_ERROR');
+    }
+
+    return output;
+  }
+
+  async openByDebt(
+    account: string,
+    portfolio: Portfolio,
+    zapToken: common.Token,
+    zapAmount: string,
+    collateralToken: common.Token,
+    debtToken: common.Token,
+    debtAmount: string,
+    slippage = defaultSlippage
+  ): Promise<OperationOutput> {
+    const output: OperationOutput = {
+      destAmount: '0',
+      afterPortfolio: portfolio.clone(),
+      logics: [],
+    };
+    try {
+      // 1. ---------- validate params ----------
+      if (debtAmount === '0') throw new OperationError('debtAmount', 'ZERO_AMOUNT');
+
+      const srcCollateral = portfolio.findSupply(collateralToken);
+      if (!srcCollateral) throw new OperationError('collateralToken', 'UNSUPPORTED_TOKEN');
+
+      const destBorrow = portfolio.findBorrow(debtToken);
+      if (!destBorrow) throw new OperationError('debtToken', 'UNSUPPORTED_TOKEN');
+
+      // 2. ---------- swap ----------
+      let supplyInput = new common.TokenAmount(collateralToken.wrapped, '0');
+      if (zapAmount !== '0' && !zapToken.wrapped.is(collateralToken.wrapped)) {
+        const swapper = this.findSwapper([zapToken, collateralToken]);
+        let swapQuotation: SwapperQuoteFields;
+        try {
+          swapQuotation = await swapper.quote({
+            input: new common.TokenAmount(zapToken, zapAmount),
+            tokenOut: collateralToken.wrapped,
+            slippage,
+          });
+        } catch {
+          throw new OperationError('zapAmount', 'NO_ROUTE_FOUND_OR_PRICE_IMPACT_TOO_HIGH');
+        }
+        const swapTokenLogic = swapper.newSwapTokenLogic(swapQuotation);
+        output.logics.push(swapTokenLogic);
+        // 2-1. the supply amount is the swap quotation output
+        supplyInput = swapQuotation.output;
+        output.afterPortfolio.supply(supplyInput.token, supplyInput.amount);
+      } else if (zapToken.wrapped.is(collateralToken.wrapped)) {
+        supplyInput = new common.TokenAmount(zapToken, zapAmount);
+        output.afterPortfolio.supply(supplyInput.token, supplyInput.amount);
+      }
+
+      // 3. leverage by debt
+      const initDebtAmount = new common.TokenAmount(destBorrow.token, destBorrow.balance);
+      const leverageAmount = new common.TokenAmount(destBorrow.token, debtAmount).sub(initDebtAmount).amount;
+      const leverageOutput = await this.leverageByDebt({
+        account,
+        portfolio,
+        srcToken: debtToken.wrapped,
+        srcAmount: leverageAmount,
+        destToken: supplyInput.token,
+      });
```

Comment:
```suggestion
      const leverageOutput = await this.leverageByDebt({
        account,
        portfolio: output.afterPortfolio,
        srcToken: debtToken.wrapped,
        srcAmount: leverageAmount,
        destToken: collateralToken, // Be consistent with `openByCollateral`
      });
```

Using `collateralToken` be consistent with `openByCollateral`